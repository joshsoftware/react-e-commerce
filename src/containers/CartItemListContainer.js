import React, { useReducer } from 'react';
import ContainerWrapper from '../components/ContainerWrapper';
import CartItem from '../components/CartItem';
import cartReducer from '../reducers/cartReducer';

const CartItemListContainer = () => {
  const [state, dispatch] = useReducer(cartReducer);
  let arr = [];
  if (state !== undefined && state.cartItemsList !== undefined) {
    arr = state.cartItemsList.map((item) => (
      <CartItem key={item.id} item={item} dispatch={dispatch} />
    ));
  }
  return <ContainerWrapper data={arr} />;
};

export default CartItemListContainer;
