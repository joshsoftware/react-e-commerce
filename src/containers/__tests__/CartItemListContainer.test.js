import React from 'react';
import CartItemListContainer from '../CartItemListContainer';
import { render } from '@testing-library/react';

describe('CartItemListContainer Component', () => {
  it('Must return column', () => {
    const { asFragment } = render(<CartItemListContainer />);

    expect(asFragment()).toMatchSnapshot();
  });
});
