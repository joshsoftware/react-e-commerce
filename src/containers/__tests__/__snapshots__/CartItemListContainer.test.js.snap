// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItemListContainer Component Must return column 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row_style row"
    >
      <div
        class="col_one col-3 col-sm-3 col-md-3 col-lg-3 col-xl-3"
      >
        <img
          alt="Product Image"
          class="img_fluid img_style card-img"
          src="https://assets.ajio.com/medias/sys_master/root/hb7/h8b/15083561943070/-473Wx593H-460422382-black-MODEL.jpg"
        />
      </div>
      <div
        class="col_two col"
      >
        <div
          class="card-title"
        >
          Football
        </div>
        <p
          class="card-text"
        >
          Price : $ 49.99
        </p>
        <p
          class="card-text"
        >
          Quantity : $ 49.99
        </p>
      </div>
      <div
        class="col_three col"
      >
        <button
          class="btn btn-secondary"
        >
          X
        </button>
      </div>
    </div>
    <div
      class="row_style row"
    >
      <div
        class="col_one col-3 col-sm-3 col-md-3 col-lg-3 col-xl-3"
      >
        <img
          alt="Product Image"
          class="img_fluid img_style card-img"
          src="https://assets.ajio.com/medias/sys_master/root/hb7/h8b/15083561943070/-473Wx593H-460422382-black-MODEL.jpg"
        />
      </div>
      <div
        class="col_two col"
      >
        <div
          class="card-title"
        >
          Baseball
        </div>
        <p
          class="card-text"
        >
          Price : $ 9.99
        </p>
        <p
          class="card-text"
        >
          Quantity : $ 9.99
        </p>
      </div>
      <div
        class="col_three col"
      >
        <button
          class="btn btn-secondary"
        >
          X
        </button>
      </div>
    </div>
    <div
      class="row_style row"
    >
      <div
        class="col_one col-3 col-sm-3 col-md-3 col-lg-3 col-xl-3"
      >
        <img
          alt="Product Image"
          class="img_fluid img_style card-img"
          src="https://assets.ajio.com/medias/sys_master/root/hb7/h8b/15083561943070/-473Wx593H-460422382-black-MODEL.jpg"
        />
      </div>
      <div
        class="col_two col"
      >
        <div
          class="card-title"
        >
          Basketball
        </div>
        <p
          class="card-text"
        >
          Price : $ 29.99
        </p>
        <p
          class="card-text"
        >
          Quantity : $ 29.99
        </p>
      </div>
      <div
        class="col_three col"
      >
        <button
          class="btn btn-secondary"
        >
          X
        </button>
      </div>
    </div>
    <div
      class="row_style row"
    >
      <div
        class="col_one col-3 col-sm-3 col-md-3 col-lg-3 col-xl-3"
      >
        <img
          alt="Product Image"
          class="img_fluid img_style card-img"
          src="https://assets.ajio.com/medias/sys_master/root/hb7/h8b/15083561943070/-473Wx593H-460422382-black-MODEL.jpg"
        />
      </div>
      <div
        class="col_two col"
      >
        <div
          class="card-title"
        >
          iPod Touch
        </div>
        <p
          class="card-text"
        >
          Price : $ 99.99
        </p>
        <p
          class="card-text"
        >
          Quantity : $ 99.99
        </p>
      </div>
      <div
        class="col_three col"
      >
        <button
          class="btn btn-secondary"
        >
          X
        </button>
      </div>
    </div>
    <div
      class="row_style row"
    >
      <div
        class="col_one col-3 col-sm-3 col-md-3 col-lg-3 col-xl-3"
      >
        <img
          alt="Product Image"
          class="img_fluid img_style card-img"
          src="https://assets.ajio.com/medias/sys_master/root/hb7/h8b/15083561943070/-473Wx593H-460422382-black-MODEL.jpg"
        />
      </div>
      <div
        class="col_two col"
      >
        <div
          class="card-title"
        >
          iPhone 5
        </div>
        <p
          class="card-text"
        >
          Price : $ 399.99
        </p>
        <p
          class="card-text"
        >
          Quantity : $ 399.99
        </p>
      </div>
      <div
        class="col_three col"
      >
        <button
          class="btn btn-secondary"
        >
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
